store/sqlstore/team_store.go

```diff
@@ -1098,20 +1098,20 @@ func (s SqlTeamStore) GetActiveMemberCount(teamId string, restrictions *model.Vi
 	query = applyTeamMemberViewRestrictionsFilterForStats(query, teamId, restrictions)
 	queryString, args, err := query.ToSql()
 	if err != nil {
-		return 0, model.NewAppError("SqlTeamStore.GetActiveMemberCount", "store.sql.build_query.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return 0, errors.Wrap(err, "team_tosql")
 	}
 
 	count, err := s.GetReplica().SelectInt(queryString, args...)
 	if err != nil {
-		return 0, model.NewAppError("SqlTeamStore.GetActiveMemberCount", "store.sql_team.get_active_member_count.app_error", nil, "teamId="+teamId+" "+err.Error(), http.StatusInternalServerError)
+		return 0, errors.Wrap(err, "failed to count TeamMembers")
 	}
 
 	return count, nil
 }
 
-func (s SqlTeamStore) GetMembersByIds(teamId string, userIds []string, restrictions *model.ViewUsersRestrictions) ([]*model.TeamMember, *model.AppError) {
+func (s SqlTeamStore) GetMembersByIds(teamId string, userIds []string, restrictions *model.ViewUsersRestrictions) ([]*model.TeamMember, error) {
 	if len(userIds) == 0 {
-		return nil, model.NewAppError("SqlTeamStore.GetMembersByIds", "store.sql_team.get_members_by_ids.app_error", nil, "Invalid list of user ids", http.StatusInternalServerError)
+		return nil, model.NewAppError("SqlTeamStore.GetMembersByIds", "app.team.get_members_by_ids.app_error", nil, "Invalid list of user ids", http.StatusInternalServerError)
```

Comment:
Why is this still returning an AppError instead of a plain error, or a custom built error?

Reply:
Good catch!

Reply:
I think I missed it because it is returning 500 when validating an input parameter, in those cases a 400 http status is mor appropriate but that would change the the implementation and could break something so, maybe I could keep it as 500 but using a plain error instead of `store.ErrInvalidInput`. what do you think @jespino ?

/cc @agnivade

Reply:
Yeah .. a simple `errors.New` sounds good to me.